kani-compiler/src/kani_middle/transform/loop_contracts.rs

```diff
@@ -0,0 +1,467 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This module contains code related to the MIR-to-MIR pass to enable loop contracts.
+//!
+
+use crate::kani_middle::KaniAttributes;
+use crate::kani_middle::find_fn_def;
+use crate::kani_middle::transform::body::{InsertPosition, MutableBody, SourceInstruction};
+use crate::kani_middle::transform::{BodyTransformation, CallGraph, TransformationResult};
+use crate::kani_queries::QueryDb;
+use crate::stable_mir::CrateDef;
+use rustc_middle::ty::TyCtxt;
+use rustc_span::Symbol;
+use stable_mir::DefId;
+use stable_mir::mir::mono::{Instance, MonoItem};
+use stable_mir::mir::{
+    BasicBlock, BasicBlockIdx, Body, ConstOperand, Operand, Rvalue, Statement, StatementKind,
+    Terminator, TerminatorKind,
+};
+use stable_mir::ty::{FnDef, MirConst, RigidTy};
+use std::collections::VecDeque;
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+
+use super::GlobalPass;
+
+/// This pass will perform the following operations:
+/// 1. Replace the body of `kani_register_loop_contract` by `kani::internal::run_contract_fn`
```

Comment:
Do we even need to invoke the loop contract at this point? I thought CBMC would handle this.